apps/src/lab2/ai/AiTutorManager.ts

```diff
@@ -0,0 +1,49 @@
+import {postAichatCompletionMessage} from '@cdo/apps/aichat/aichatApi';
+import {PendingChatMessage, AichatContext} from '@cdo/apps/aichat/types';
+import {EMPTY_AI_CUSTOMIZATIONS} from '@cdo/apps/aichat/views/modelCustomization/constants';
+import {Role} from '@cdo/apps/aiComponentLibrary/chatMessage/types';
+import {getStore} from '@cdo/apps/redux';
+import {
+  AiChatModelIds,
+  AiInteractionStatus as Status,
+} from '@cdo/generated-scripts/sharedConstants';
+
+const systemPrompts = {
+  hint: "You are responding to a query about programming.  Target the reading age of an American 7th grader.  Use the Socratic method to guide the student to the answer, but do not give them the answer directly.  Just focus on the biggest single issue you find.  Use plain English in the answer.  I don't want multiple steps, points, or questions.  Just one question that helps the student to make progress.",
+  user: 'You are responding to a query about programming.  Target the reading age of an American 7th grader.  Use plain English in the answer.  Keep the answer relatively short, say three or fewer paragraphs, with each paragraph three sentences or less.',
+};
+
+export default class AiTutorManager {
+  async askAiTutor(message: string, type: 'hint' | 'user') {
+    const state = getStore().getState();
```

Comment:
minor: can we pass in the relevant state info to avoid having to reference the redux store here?

Reply:
Good call.  I've hoisted the retrieval up to `AiTutorUi` for now, but if we want that component to be used in legacy labs as well, we'll probably have to hoist the retrieval a bit higher again.